packages/cli-lib/lib/lang.js

```diff
@@ -0,0 +1,79 @@
+const util = require('util');
+const path = require('path');
+const fs = require('fs-extra');
+const handlebars = require('handlebars');
+const yaml = require('js-yaml');
+const { logger } = require('../logger');
+
+const MISSING_LANGUAGE_DATA_PREFIX = '[Missing language data]';
+
+let languageObj;
+
+const loadLanguageFromYaml = () => {
+  if (languageObj) return;
+
+  try {
+    languageObj = yaml.load(
+      fs.readFileSync(path.join(__dirname, '../lang/en.lyaml'), 'utf8')
+    );
+    logger.debug(
+      'Loaded language data: ',
+      util.inspect(languageObj, true, 999, true)
+    );
+  } catch (e) {
+    logger.error('Error loading language data: ', e);
+  }
+};
+
+const getTextValue = lookupDotNotation => {
+  const lookupProps = lookupDotNotation.split('.');
+  const missingTextData = `${MISSING_LANGUAGE_DATA_PREFIX}: ${lookupDotNotation}`;
+  let textValue = languageObj;
+  let previouslyCheckedProp = lookupProps[0];
+
+  try {
+    lookupProps.forEach(prop => {
+      textValue = textValue[prop];
+      previouslyCheckedProp = prop;
+    });
+  } catch (e) {
+    logger.error(
+      `Unable to access language property: ${lookupDotNotation}. Failed to access prop "${previouslyCheckedProp}".`
+    );
+    return missingTextData;
+  }
+
+  if (!textValue) {
+    return missingTextData;
+  }
+
+  return textValue;
+};
+
+const getInterpolatedValue = (textValue, interpolationData) => {
+  const template = handlebars.compile(textValue);
+
+  return template(interpolationData);
+};
+
+const i18n = (lookupDotNotation, options = {}) => {
+  if (typeof lookupDotNotation !== 'string') {
+    throw new Error(
+      `i18n must be passed a string value for lookupDotNotation, received ${typeof lookupDotNotation}`
+    );
+  }
+
+  const textValue = getTextValue(lookupDotNotation);
+  const shouldInterpolate =
+    options.data && !textValue.startsWith(MISSING_LANGUAGE_DATA_PREFIX);
```

Comment:
I think `tokens` might be a clearer name than `data`.

Reply:
Thanks for pointing this out, I think `data` is too generic here. I thought I was using the property name that i18n was using, but now I can't verify that. I'm not sure why I nested values within `data` -- I think it may be better and more consistent with i18n to remove the nesting of the values under the `data` property all together.